### DISCUSSION TITLE
Mocking Prisma Guide(Client): Script `prisma` import

### DISCUSSION THREAD

**Discussion by papsavas** [prisma/prisma#18882] [Q&A] [answered]
> *referring to [this blog post](https://www.prisma.io/blog/testing-series-1-8eRB5p0Y8o) by @sabinadams* 

In this example, is `currentUser` supposed to *ever* perform a DB write? 
- Yes: 
  ❔ Is its closure supposed to reference the mocked `prisma` in testing and avoid actual writes?. Because this is not happening in my case

- No:
  ❗Imported `prisma` should be the mocked one.

![image](https://user-images.githubusercontent.com/50584606/233806628-41bf4b47-5514-4981-8135-6180e8ad9a41.png)
![image](https://user-images.githubusercontent.com/50584606/233806648-8bd1ada0-9948-46d1-b4a4-4e23746f2e0c.png)

**Comment by nurul3101** [comment, marked as answer]
Hello @papsavas 👋 

At first, the `createUser` function actually calls the database (so yes, the method performs a DB write). It is using the real client instance to demonstrate the problem you may run into when testing against a real database.

After demonstrating this, later we use the mocked prisma client instance to not perform a DB operation for test.

**Reply by papsavas** [reply]
I Believe this should be a highlighted change in scripts code. The last visible script import is the real client instance. After that there are only commented lines
![image](https://user-images.githubusercontent.com/50584606/233985873-a9110cfd-6482-4c1a-978e-75a7ef970cd4.png)

**Reply by nurul3101** [reply]
Thanks for sharing the feedback, I'll share it internally.
